lib/errors.js

```diff
@@ -1,7 +1,16 @@
 module.exports = class HypercoreError extends Error {
   constructor (msg, code) {
-    super(msg)
+    super(`${code}: ${msg}`)
     this.code = code
+
+    if (Error.captureStackTrace) {
+      const ctor = this.constructor
+      Error.captureStackTrace(this, ctor[code] || ctor)
+    }
```

Comment:
what does this do? remove the first line in the stack?

Reply:
It removes the error constructor from the stack trace, yes. It's the `at Function.BAD_ARGUMENT (lib/errors.js:8:12)` bit in the description above.

Reply:
Ah ok, I wanna research if that has any implications for perf (stacks are expensive), as we make more errors now due to the SNAPSHOT one. Could we move the stack stuff to a new PR, the rest is def killer.

Reply:
There's an overhead of ~11 µs on my hardware for each call, which is about the same overhead of constructing an `Error` object. It therefore effectively doubles the overhead.

Reply:
Ah so basically free, that's fine

Reply:
instead of constructor[ctor], pass the function down. ie

```js
static BAD_ARGUMENT (msg) {
  return new HypercoreError(msg, 'BAD_ARGUMENT', this.BAD_ARGUMENT)
}
```

Reply:
then I'm happy :D

Reply:
Perfect, I'll tweak 👍